cmd/utils.go

```diff
@@ -1306,17 +1306,21 @@ func GenDeploymentYaml(log *LoggingConfig, appName string, imageName string, con
 	//Set the containerPort
 	containerPorts := make([]*Port, 0)
 	for i, port := range ports {
-		//KNative only allows a single port entry
-		if i == 0 {
-			yamlMap.Spec.PodTemplate.Spec.Containers[0].Ports = containerPorts
-		}
-		log.Debug.Log("Adding port to yaml: ", port)
-		newContainerPort := new(Port)
-		newContainerPort.ContainerPort, err = strconv.Atoi(port)
-		if err != nil {
-			return "", err
+		if port == debugPort {
+			log.Debug.log("Debug port found. Skipping addition to the deployment file...")
+		} else {
+			//KNative only allows a single port entry
+			if i == 0 {
+				yamlMap.Spec.PodTemplate.Spec.Containers[0].Ports = containerPorts
+			}
+			log.Debug.Log("Adding port to yaml: ", port)
+			newContainerPort := new(Port)
+			newContainerPort.ContainerPort, err = strconv.Atoi(port)
+			if err != nil {
+				return "", err
+			}
+			yamlMap.Spec.PodTemplate.Spec.Containers[0].Ports = append(yamlMap.Spec.PodTemplate.Spec.Containers[0].Ports, newContainerPort)
```

Comment:
I believe this is currently failing if we have multiple debug ports, because this code will get executed for all but one of the multiple debug ports.